install/helm/agones/templates/extensions.yaml

```diff
@@ -11,11 +11,22 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-{{- $ca := genCA "admission-controller-ca" 3650 }}
-{{- $cn := printf "agones-controller-service" }}
-{{- $altName1 := printf "agones-controller-service.%s"  .Release.Namespace }}
-{{- $altName2 := printf "agones-controller-service.%s.svc" .Release.Namespace }}
-{{- $cert := genSignedCert $cn nil (list $altName1 $altName2) 3650 $ca }}
+{{- $cert := "" }}
+{{- $key := "" }}
+{{- if .Values.agones.controller.generateTLS }}
+  {{- $previous := lookup "v1" "Secret" .Release.Namespace (include "agones.fullname" . | printf "%s-cert") }}
+  {{- if $previous | and .Values.agones.controller.reuseGeneratedTLS }}
+    {{- $cert = index $previous.data "server.crt" | b64dec }}
+    {{- $key = index $previous.data "server.key" | b64dec }}
+  {{- else }}
+    {{- $cn := printf "agones-controller-service.%s.svc" .Release.Namespace }}
+    {{- $altName1 := printf "agones-controller-service.%s"  .Release.Namespace }}
+    {{- $altName2 := printf "agones-controller-service.%s.svc" .Release.Namespace }}
+    {{- $certObj := genSelfSignedCert $cn nil (list $altName1 $altName2) 3650 }}

```

Comment:
Maybe you already answered this (it's been a while since I reviewed this PR), but can you explain why this isn't using the `genCA` command any longer?

To @markmandel's earlier review comment - ideally after this change nothing looks different than it did before _unless_ you set the new helm value. From what I can tell here, this changes the way that certs are generated even if you _don't_ set the new value to true.

Reply:
Like I said in this comment (https://github.com/googleforgames/agones/pull/2417#discussion_r798486250) - there is difference with approach in this helm file and `install/helm/agones/certs/cert.sh`.
 
After my change nothing looks different in terms of how parts of this helm chart works. 
The only difference is that I use self signed certificate feature (just like certificate generated by `install/helm/agones/certs/cert.sh`) that it can be its own certification authority - that's why it is used in place of `caBundle`.  This change also unify usage of certs generated by helm and by mentioned above script.